src/makeCalculator.js

```diff
@@ -36,8 +36,41 @@
  *
  * @return {object}
  */
+
 function makeCalculator() {
-  // write code here
+  const calc = {
+    result: 0,
+
+    addition(a, b) {
+      return a + b;
+    },
+
+    subtraction(a, b) {
+      return a - b;
+    },
+
+    division(a, b) {
+      return a / b;
+    },
+
+    multiplication(a, b) {
+      return a * b;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(operation, number) {
+      this.result = operation(this.result, number);
```

Comment:
do you need to pass `this` explicitly here? To me it seems that your methods already have an access to `this`